porcupine/plugins/anchors.py

```diff
@@ -0,0 +1,173 @@
+        anchor_list = self._get_anchors()
+        anchor_rows = sorted(
+            [
+                int(self.tab_textwidget.index(anchorpoint).split(".")[0])
+                for anchorpoint in anchor_list
+            ]
+        )
+
+        next_anchor_row = 0
+        for anchor_row in anchor_rows:
+            if anchor_row < int(cursor_row):
+                next_anchor_row = anchor_row
+            else:
+                break
+
+        if not next_anchor_row:
+            return
+        else:
+            self.tab_textwidget.mark_set("insert", f"{str(next_anchor_row)}.0")
+
+        # If cursor is above first row
+        if int(self._get_cursor_index().split(".")[0]) < int(
+            self.tab_textwidget.index("@0,0").split(".")[0]
+        ):
+            self.tab_textwidget.see("insert")
+
+        # TODO: Current Bugs:
+        # 1) Jump to previous will delete code on current row if no more anchors above.
+        # 2) Can't commit/push because of git merge weirdness
+
+    # def bind_specific(self, event: tkinter.Event[tkinter.Misc], partial ?) -> None:
+    #     pass
+
+    # def jump_to_specific(self, event: tkinter.Event[tkinter.Misc], partial ?) -> None:
+    #     pass
+
+    def update_linenumbers(self, event: tkinter.Event[tkinter.Misc]) -> None:
+        """
+        Re-draws the anchor points every time the linenumber instance updates
+        (scroll, insertion/deletion of text)
+        """
+        self.prevent_duplicate_anchors()
+
+        anchor_list = self._get_anchors()
+
+        for anchorpoint in anchor_list:
+            row_tag = "line_" + self.tab_textwidget.index(anchorpoint).split(".")[0]
+            try:
+                [row_id] = self.linenumbers.find_withtag(row_tag)
+                row_text = self.linenumbers.itemcget(row_id, "text")
+                self.linenumbers.itemconfigure(row_id, text=row_text + " " + self.anchor_symbol)
+            except ValueError:
+                pass
+
+    def prevent_duplicate_anchors(self) -> None:
+        current_rows = set()
+        anchor_list = self._get_anchors()
+
+        for anchorpoint in anchor_list:
+            anchor_row = self.tab_textwidget.index(anchorpoint).split(".")[0]
+            if anchor_row in current_rows:
+                self.tab_textwidget.mark_unset(anchorpoint)
+            else:
+                self.tab_textwidget.mark_set(anchorpoint, f"{anchorpoint} linestart")
+                current_rows.add(anchor_row)
+
+
+def on_new_tab(tab: tabs.Tab) -> None:
+    if isinstance(tab, tabs.FileTab):
+        [linenumbers] = [
+            child for child in tab.left_frame.winfo_children() if isinstance(child, LineNumbers)
+        ]
+        anchor = AnchorManager(tab.textwidget, linenumbers)
+        tab.textwidget.bind("<Control-g>", anchor.toggle_on_off, add=True)
+        tab.textwidget.bind("<Control-k>", anchor.jump_to_previous, add=True)
+        tab.textwidget.bind("<Control-l>", anchor.jump_to_next, add=True)
```

Comment:
We should make these configurable in `default_keybindings.tcl`. See other plugins for examples.

Reply:
These are still subject for change. Will add todo comment